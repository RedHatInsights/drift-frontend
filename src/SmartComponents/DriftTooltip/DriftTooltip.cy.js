import { mount } from '@cypress/react';
import React from 'react';
import DriftTooltip from './DriftTooltip';

describe('DriftTooltip', () => {
    it('renders correctly', () => {
        mount(<DriftTooltip content={ <p>Hidden text</p> } body={ <span>Text</span> }/>);

        cy.get('span').click();
        cy.get('.pf-c-tooltip').should('have.text', 'Hidden text');
    });
});
