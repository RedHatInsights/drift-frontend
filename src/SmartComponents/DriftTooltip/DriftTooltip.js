import React from 'react';
import PropTypes from 'prop-types';
import { Tooltip } from '@patternfly/react-core';

export const DriftTooltip = ({
    content,
    body
}) => {
    return (
        <Tooltip
            content={ content }
        >
            { body }
        </Tooltip>
    );
};

DriftTooltip.propTypes = {
    content: PropTypes.string,
    body: PropTypes.any
};

export default DriftTooltip;
