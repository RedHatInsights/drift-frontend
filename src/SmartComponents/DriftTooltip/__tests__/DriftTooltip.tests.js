import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { PlaystationIcon } from '@patternfly/react-icons';

import DriftTooltip from '../DriftTooltip';

describe('DriftTooltip', () => {
    let props;

    beforeEach(() => {
        props = {
            content: 'This is a tooltip.',
            body: <PlaystationIcon />
        };
    });

    it('should render correctly', () => {
        const wrapper = shallow(
            <DriftTooltip { ...props } />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
