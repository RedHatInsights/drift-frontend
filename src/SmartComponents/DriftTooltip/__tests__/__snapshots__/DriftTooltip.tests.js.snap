// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DriftTooltip should render correctly 1`] = `
<Tooltip
  content="This is a tooltip."
>
  <PlaystationIcon
    color="currentColor"
    noVerticalAlign={false}
    size="sm"
  />
</Tooltip>
`;
